db/src/db.rs

```diff
@@ -27,8 +28,23 @@ use mentat_tx::entities as entmod;
 use mentat_tx::entities::{Entity, OpType};
 use types::*;
 
-pub fn new_connection() -> rusqlite::Connection {
-    return rusqlite::Connection::open_in_memory().unwrap();
+pub fn new_connection<T>(uri: T) -> rusqlite::Result<rusqlite::Connection> where T: AsRef<Path> {
+    let res = match uri.as_ref().to_string_lossy().len() {
+        0 => rusqlite::Connection::open_in_memory(),
+        _ => rusqlite::Connection::open(uri),
+    };
+
+    if let Ok(conn) = res.as_ref() {
```

Comment:
This would be simpler, I think, if it was like
```rust
fn (..) {
  let conn = match {
    _ => ... open(uri)?
  };
  conn.execute_batch()
}
```
The `as_ref` with a `Result` threw me a little.